src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1084,6 +1084,36 @@ impl ToQemuParams for DeviceVirtconsole {
     }
 }
 
+#[derive(Debug)]
+struct DeviceIntelIommu {
+    intremap: bool,
+    device_iotlb: bool,
+    caching_mode: bool,
+}
+
+impl DeviceIntelIommu {
+    fn new() -> DeviceIntelIommu {
+        DeviceIntelIommu {
+            intremap: true,
+            device_iotlb: true,
+            caching_mode: true,
+        }
+    }
+}
+
+#[async_trait]
+impl ToQemuParams for DeviceIntelIommu {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut params = Vec::new();
+        params.push("intel-iommu".to_owned());
+        let to_onoff = |b| if b { "on" } else { "off" };
```

Comment:
It seems that the `else` branch of `to_onoff` is never reached ? as the result of all fields of `DeviceIntelIommu` initialized with `true`.

Reply:
Yes, you're right.  As you can see, the whole `DeviceIntelIommu` struct is a bit make-believe since the initialisation is basically hard-coded, just like in the go runtime.  Here I just preferred to limit the hard-coded bit to the constructor and write the rest of the class _as if_ the values weren't hard-coded.

It might help just a little bit if we stop hard-coding the initialisation at some point in the future...

Reply:
Ok SGTM !